### ISSUE TITLE
Global user defined rate limit

### ISSUE THREAD

**Issue by papapumpnz** [laurentS/slowapi#37]
Routes without a decorator are not being limited when using a global defined rate limit.

**Comment by papapumpnz** [comment]
I see the <code>application_limits</code> callable on the Limiter class which I have defined, but routes without a @limiter decorator on the endpoint it is not being rate limited.

```
limiter = Limiter(key_func=default_identifier,
                  application_limits=[get_user_rate_limit],
                  headers_enabled=True,
                  storage_uri=settings.RateLimit.redis_uri,
                  in_memory_fallback_enabled=True,
                  swallow_errors=False)
app.state.limiter = limiter
app.add_exception_handler(RateLimitExceeded, _rate_limit_exceeded_handler)
app.add_middleware(SlowAPIMiddleware)
```

```
def get_user_rate_limit(request: Request) -> str:
    """
    Returns a rate limit string based on user role if known, else returns a default
    :return str: a rate limit string
    """
    return "100/1 day"
```

**Comment by papapumpnz** [comment]
@laurentS I think this is the cause of the bug above.

middleware:SlowAPIMiddleware in line 25:
```
if  match.FULL and hasattr(route, "endpoint"):
```
should be:
```
if match is match.FULL and hasattr(route, "endpoint"):
```
Seems that match.FULL enum is not be compared as true and is always matching, and then all routes are set as the handler and whatever route is last in the routes list gets picked.

Making the change seems to fix the problem at my end.

**Comment by laurentS** [comment]
@papapumpnz that looks a lot like the fix in https://github.com/laurentS/slowapi/pull/30 does the PR work for you as well?

**Comment by papapumpnz** [comment]
@laurentS  Yes, tested branch and it does.

**Comment by laurentS** [comment]
Fixed via PR #30  merged and released in 0.1.4
